xir/program.py

```diff
@@ -507,7 +535,13 @@ def add_observable(
                 f"Observable '{name}' already defined."
                 "Replacing old definition with new definition."
             )
-
+        if wires is None:
+            wires = []
+            for statement in statements:
+                for factor in statement.factors:
+                    for wire in factor.wires:
+                        if wire not in wires:
+                            wires.append(wire)
```

Comment:
Are you sure this is needed? I thought we accounted for this in `Transformer.obs_def`.